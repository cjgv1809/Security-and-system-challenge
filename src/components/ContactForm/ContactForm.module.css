.contactForm {
  padding: 1.5rem;
  margin: 0 auto 4rem;
  background: var(--color-background-light);
  border-radius: 16px;
  box-shadow: 0 4px 30px var(--color-box-shadow);
  backdrop-filter: blur(6px);
  -webkit-backdrop-filter: blur(6px);
  border: 1px solid var(--color-border-light);
  width: min(100%, 600px);
}

.title {
  font-size: 1.5rem;
  margin-bottom: 1.25rem;
}

.form {
  display: flex;
  flex-direction: column;
}

.field {
  display: flex;
  flex-direction: column;
  margin-bottom: 0.9375rem;
}

.label {
  font-size: 1rem;
  font-weight: 600;
  margin-bottom: 0.3125rem;
}

.input,
.textarea {
  margin-bottom: 0.9375rem;
  padding: 0.625rem;
  border: 1px solid var(--color-input-border);
  border-radius: 6px;
  background-color: var(--color-input-background);
  color: var(--color-input-text);
  font-size: 1rem;
}

.textarea {
  resize: none;
}

:is(.input, .textarea)::placeholder {
  color: var(--color-placeholder-text);
}

.helperText {
  font-size: 0.8rem;
  color: var(--color-item-text);
  margin: 0 0 0.625rem;
}

.button {
  padding: 0.625rem 1.25rem;
  background-color: var(--color-button-background);
  color: var(--color-active-text);
  border: none;
  border-radius: 6px;
  font-size: 1rem;
  font-weight: 700;
  cursor: pointer;
  transition: background-color 0.3s ease-in-out;
}

.button:hover {
  background-color: var(--color-button-hover);
}
