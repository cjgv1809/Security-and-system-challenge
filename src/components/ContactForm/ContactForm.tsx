"use client";

import React, { useRef, useState } from "react";
import styles from "./ContactForm.module.css";
import { submitForm } from "@/actions/submitForm";
import Alert from "../Alert/Alert";
import type { AlertType } from "@/app/types";

const ContactForm: React.FC = () => {
  const [alert, setAlert] = useState<{
    type: AlertType;
    message: string;
  } | null>(null);
  const formRef = useRef<HTMLFormElement>(null);

  const handleSubmit = async (formData: FormData) => {
    const formFields = {
      name: formData.get("name") as string,
      email: formData.get("email") as string,
      message: formData.get("message") as string,
    };

    try {
      const result = await submitForm(formFields);
      console.log(result);

      if (result.success) {
        setAlert({ type: "success", message: "Mensaje enviado exitosamente" });
      } else {
        setAlert({
          type: "error",
          message: result.errors?.message[0] || "Error al enviar el formulario",
        });
      }
      formRef.current?.reset();
    } catch (error) {
      setAlert({
        type: "error",
        message: "Error inesperado al enviar el formulario",
      });
      formRef.current?.reset();
    }
  };

  return (
    <section className={styles.contactForm}>
      <h2 className={styles.title}>Contacto</h2>

      {alert && (
        <Alert
          type={alert.type}
          message={alert.message}
          onClose={() => setAlert(null)}
        />
      )}

      <form action={handleSubmit} className={styles.form} ref={formRef}>
        <div className={styles.field}>
          <label htmlFor="name" className={styles.label}>
            Nombre
          </label>
          <input
            id="name"
            type="text"
            name="name"
            placeholder="Ingresa tu nombre"
            className={styles.input}
            required
          />
        </div>
        <div className={styles.field}>
          <label htmlFor="email" className={styles.label}>
            Email
          </label>
          <input
            id="email"
            type="email"
            name="email"
            placeholder="Ingresa tu email"
            className={styles.input}
            required
          />
        </div>
        <div className={styles.field}>
          <label htmlFor="message" className={styles.label}>
            Mensaje
          </label>
          <textarea
            id="message"
            name="message"
            placeholder="Escribe tu mensaje"
            className={styles.textarea}
            rows={5}
            required
          ></textarea>
          <p className={styles.helperText}>
            El mensaje debe contener al menos 10 caracteres
          </p>
        </div>
        <button type="submit" className={styles.button}>
          Enviar
        </button>
      </form>
    </section>
  );
};

export default ContactForm;
