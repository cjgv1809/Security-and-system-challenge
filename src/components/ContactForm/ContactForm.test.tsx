import React from "react";
import { render, screen } from "@testing-library/react";
import ContactForm from "./ContactForm";

describe("ContactForm Component", () => {
  it("should render the form correctly", () => {
    render(<ContactForm />);
    expect(screen.getByLabelText(/nombre/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/email/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/mensaje/i)).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /enviar/i })).toBeInTheDocument();
  });

  it("should match the snapshot", () => {
    const { container } = render(<ContactForm />);
    expect(container).toMatchSnapshot();
  });
});
