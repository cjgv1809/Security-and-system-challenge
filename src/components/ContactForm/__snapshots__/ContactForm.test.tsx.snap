// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm Component should match the snapshot 1`] = `
<div>
  <section
    class="contactForm"
  >
    <h2
      class="title"
    >
      Contacto
    </h2>
    <form
      class="form"
    >
      <div
        class="field"
      >
        <label
          class="label"
          for="name"
        >
          Nombre
        </label>
        <input
          class="input"
          id="name"
          name="name"
          placeholder="Ingresa tu nombre"
          required=""
          type="text"
        />
      </div>
      <div
        class="field"
      >
        <label
          class="label"
          for="email"
        >
          Email
        </label>
        <input
          class="input"
          id="email"
          name="email"
          placeholder="Ingresa tu email"
          required=""
          type="email"
        />
      </div>
      <div
        class="field"
      >
        <label
          class="label"
          for="message"
        >
          Mensaje
        </label>
        <textarea
          class="textarea"
          id="message"
          name="message"
          placeholder="Escribe tu mensaje"
          required=""
          rows="5"
        />
        <p
          class="helperText"
        >
          El mensaje debe contener al menos 10 caracteres
        </p>
      </div>
      <button
        class="button"
        type="submit"
      >
        Enviar
      </button>
    </form>
  </section>
</div>
`;
